terraform/node_resource_plan_orphan.go

```diff
@@ -48,6 +49,14 @@ func (n *NodePlannableResourceInstanceOrphan) Execute(ctx EvalContext, op walkOp
 	}
 }
 
+func (n *NodePlannableResourceInstanceOrphan) ProvidedBy() (addr addrs.ProviderConfig, exact bool) {
+	if n.Addr.Resource.Resource.Mode == addrs.DataResourceMode {
+		// indicate that this node does note require a configured provider
+		return nil, true
```

Comment:
"This resource is provided by exactly no provider" is a funny sentence to think about :grinning: but agreed that this feels like the most pragmatic answer within our current design.

I think a general learning I've had navigating this problem several times now is first that optional interfaces are non-ideal and we should avoid them where possible, but also that when we _do_ end up using them there should always be a way to dynamically opt-out, as well as statically opt-out, because there does always seem to end up being some situation like this where the decision ends up being made dynamically.

Reply:
heh, stating out the implied meaning in that way does sound strange, but the primary factor in being _exact_ is that it avoids the possibility of the zero values indicating this specific condition ;)